### PR TITLE
Be clearer about current orphaned channel semantics

### PR DESCRIPTION
**Pull request by daviddrysdale**
Notwithstanding the discussion in #763, this PR tests and documents current semantics.

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [x] I have written tests that cover the code changes.
  - [x] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: examples/abitest/module_0/rust/src/lib.rs [hunk 1]
```diff
@@ -755,7 +771,20 @@ impl FrontendNode {
             Err(OakStatus::ErrChannelClosed),
             oak::channel_write(out_channel, &data, &[])
         );
+        expect_eq!(Ok(()), oak::channel_close(out_channel.handle));
 
+        // The same is true even if there are messages pending on the channel.
```
Comment (tiziano88) [inline examples/abitest/module_0/rust/src/lib.rs:776]: This seems a separate test case from the previous one, worth splitting? up to you

File: examples/abitest/module_0/rust/src/lib.rs [hunk 2]
```diff
@@ -755,7 +771,20 @@ impl FrontendNode {
...
+        let (out_channel, in_channel) = oak::channel_create().unwrap();
```
Comment (tiziano88) [inline examples/abitest/module_0/rust/src/lib.rs:777]: I think read and write are clearer than in and out, but maybe it's just me.

File: examples/abitest/module_0/rust/src/lib.rs (hunk 1 above)
Comment (daviddrysdale) [reply examples/abitest/module_0/rust/src/lib.rs]: Yep, split.

File: examples/abitest/module_0/rust/src/lib.rs (hunk 2 above)
Comment (daviddrysdale) [reply examples/abitest/module_0/rust/src/lib.rs:784]: Probably yes, but at this point it feels better to be consistent with all the other examples.

**Pull request merged**